schema.ts

```diff
@@ -0,0 +1,307 @@
+   * self-registering element.
+   * 
+   * Self-registering elements must also include a CustomElementDefintionDoc
+   * in the module's exports.
+   */
+  tagName?: string;
+
+  /**
+   * The attributes that this element is known to understand.
+   */
+  attributes?: AttributeDoc[];
+
+  /** The events that this element fires. */
+  events?: EventDoc[];
+
+  /**
+   * The shadow dom content slots that this element accepts.
+   */
+  slots?: SlotDoc[];
+
+  demos?: Demo[];
+}
+
+export interface AttributeDoc {
+  name: string;
+
+  /**
+   * A markdown description for the attribute.
+   */
+  description?: string;
+
+  /**
+   * The type that the attribute will be serialized/deserialized as.
+   */
+  type?: string;
+
+  /**
+   * The default value of the attribute, if any.
+   *
+   * As attributes are always strings, this is the actual value, not a human
+   * readable description.
+   */
+  defaultValue?: string;
+
+  /**
+   * The name of the field this attribute is associated with, if any.
+   */
+  fieldName?: string;
+}
+
+export interface EventDoc {
+  name: string;
+
+  /**
+   * A markdown description of the event.
+   */
+  description?: string;
+
+  /**
+   * The type of the event object that's fired.
+   *
+   * If the event type is built-in, this is a string, e.g. `Event`,
+   * `CustomEvent`, `KeyboardEvent`. If the event type is an event class defined
+   * in a module, the reference to it.
+   */
+  type: Reference|string;
+
+  /**
+   * If the event is a CustomEvent, the type of `detail` field.
+   */
+  detailType?: string;
+}
+
+export interface SlotDoc {
+  /**
+   * The slot name, or the empty string for an unnamed slot.
+   */
+  name: string;
+
+  /**
+   * A markdown description of the part.
+   */
+  description?: string;
+}
+
+/**
+ * The description of a CSS Part
+ */
+export interface CssPartDoc {
+  name: string;
+
+  /**
+   * A markdown description for the CSS property.
+   */
+  description?: string;
+}
+
+export interface CssCustomPropertyDoc {
+  /**
+   * The name of the property, including leading `--`.
+   */
+  name: string;
+
+  defaultValue?: string;
+
+  /**
+   * A markdown description for the attribute.
+   */
+  description?: string;
+}
+
+export interface ClassDoc {
+  kind: 'class';
+  name: string;
+
+  /**
+   * A markdown summary suitable for display in a listing.
+   * TODO: restrictions on markdown/markup. ie, no headings, only inline
+   *       formatting?
+   */
+  summary?: string;
+
+  /**
+   * A markdown description of the class.
+   */
+  description?: string;
+  superclass?: Reference;
+  mixins?: Array<Reference>;
+  members?: Array<ClassMember>;
+}
+
+export type ClassMember = FieldDoc|MethodDoc;
+
+export interface FieldDoc {
+  kind: 'field';
+  name: string;
+  static?: boolean;
+
+  /**
+   * A markdown summary suitable for display in a listing.
+   * TODO: restrictions on markdown/markup. ie, no headings, only inline
+   *       formatting?
+   */
+  summary?: string;
```

Comment:
Is present in WCA output, but not in the schema

`default?: string;`

Reply:
added